cpp/serve/radix_tree.h

```diff
@@ -0,0 +1,417 @@
+/*!
+ *  Copyright (c) 2023 by Contributors
+ * \file serve/radix_tree.h
+ */
+#ifndef MLC_LLM_SERVE_RADIX_TREE_H_
+#define MLC_LLM_SERVE_RADIX_TREE_H_
+#include <tvm/runtime/container/shape_tuple.h>
+#include <tvm/runtime/object.h>
+
+#include <unordered_map>
+#include <unordered_set>
+
+namespace mlc {
+namespace llm {
+namespace serve {
+
+using namespace tvm::runtime;
+
+/*!
+ * \brief The sequence ID linked list structure in paged radix tree node.
+ */
+struct SequenceIDNode {
+  /*! \brief The stored sequence ID. */
+  int64_t id;
+  /*! \brief The pointer to the next sequence ID. */
+  SequenceIDNode* next;
+};
+
+/*!
+ * \brief The sequence Id node pool.
+ *
+ * The sequence Id node pool allocates all sequence ID nodes when construction and frees when
+ * destruction, to avoid frequent memory operation.
+ */
+class SequenceIDNodePool {
+ private:
+  /*! \brief The number of nodes in sequence ID node pool. */
+  size_t num_nodes;
```

Comment:
The name of private members should end with a underscore

And put private members at the end of the class